crates/autopilot/src/run_loop.rs

```diff
@@ -368,17 +374,31 @@ impl RunLoop {
             .solutions
             .into_iter()
             .map(|solution| {
-                Ok(Solution {
-                    id: solution.solution_id,
-                    account: solution.submission_address,
-                    score: NonZeroU256::new(solution.score).ok_or(ZeroScoreError)?,
-                    orders: solution
-                        .orders
-                        .into_iter()
-                        .map(|(o, amounts)| (o.into(), amounts))
-                        .collect(),
-                    clearing_prices: solution.clearing_prices,
-                })
+                let mut prices = HashMap::new();
+                for (token, price) in solution.clearing_prices.into_iter() {
+                    prices.insert(token.into(), auction::Price::new(price.into())?);
+                }
+                let orders = solution
+                    .orders
+                    .into_iter()
+                    .map(|(o, amounts)| {
+                        (
+                            o.into(),
+                            competition::TradedAmounts {
+                                sell: amounts.sell_amount.into(),
+                                buy: amounts.buy_amount.into(),
+                            },
+                        )
+                    })
+                    .collect();
+
+                Ok(competition::Solution::new(
+                    solution.solution_id,
+                    solution.submission_address.into(),
+                    NonZeroU256::new(solution.score).ok_or(ZeroScoreError)?,
```

Comment:
If we have a constructor, why does it not have this logic (seems a bit useless to have `new` just for variable assignments)

Reply:
I've moved all of this code to dto conversion as it should be: https://github.com/cowprotocol/services/pull/2736

Reply:
The explicit reason why it's not inside `new()` is because `new()` should not accept non-domain objects as parameters. Instead, I moved the data conversion to `into_domain` function on the dto object itself.